### PR TITLE
refactor(PN-11896): rework on courtesyAddressContacts

### PR DESCRIPTION
**Pull request by SarahDonvito**
## Short description
Rework of common components and courtesy addresses for PF and PG.

## List of changes proposed in this pull request
- Removed `actions` prop and use of `memo` in DigitalContactsCard component
- Catched error in IOContact component (only PF)
- Moved content of CourtesyContactsList in CourtesyContacts component

## How to test
Login as PF/PG; check in contacts page UX/UI is not changed, but components work correctly.

### CODE REVIEW

File: packages/pn-personafisica-webapp/src/components/Contacts/__test__/CourtesyContacts.test.tsx [hunk 1]
```diff
@@ -88,6 +88,43 @@ describe('CourtesyContacts Component', async () => {
     expect(emailInput).toBeInTheDocument();
   });
 
+  it('renders components - contacts', async () => {
+    await act(async () => {
+      result = render(
+        <DigitalContactsCodeVerificationProvider>
```
Comment (AndreaCimini90) [inline packages/pn-personafisica-webapp/src/components/Contacts/__test__/CourtesyContacts.test.tsx:94]: replace with

```
 <DigitalContactsCodeVerificationProvider>
          <CourtesyContacts contacts={digitalCourtesyAddresses} />
        </DigitalContactsCodeVerificationProvider>
```

File: packages/pn-personafisica-webapp/src/components/Contacts/__test__/CourtesyContacts.test.tsx [hunk 2]
```diff
@@ -88,6 +88,43 @@ describe('CourtesyContacts Component', async () => {
...
+    expect(buttons![1]).toBeEnabled();
+    expect(buttons![0].textContent).toMatch('button.modifica');
+    expect(buttons![1].textContent).toMatch('button.elimina');
+    expect(buttons![2]).toBeEnabled();
+    expect(buttons![3]).toBeEnabled();
+    expect(buttons![2].textContent).toMatch('button.modifica');
+    expect(buttons![3].textContent).toMatch('button.elimina');
+  });
+
   it('add new phone number', async () => {
```
Comment (AndreaCimini90) [inline packages/pn-personafisica-webapp/src/components/Contacts/__test__/CourtesyContacts.test.tsx:128]: In this test and below we are using CourtesyContactItem, but it is wrong because we are testing CourtesyContacts component. So rework the tests replacing CourtesyContactItem with CourtesyContacts

**Review by AndreaCimini90 — CHANGES_REQUESTED** [review]

File: packages/pn-personafisica-webapp/src/components/Contacts/__test__/CourtesyContacts.test.tsx [hunk 3]
```diff
@@ -176,16 +213,16 @@ describe('CourtesyContacts Component', async () => {
...
       </DigitalContactsCodeVerificationProvider>,
       {
         preloadedState: {
           contactsState: { digitalAddresses: [defaultPhoneAddress] },
         },
       }
     );
-    const editButton = result.getByRole('button', { name: 'button.modifica' });
-    fireEvent.click(editButton);
+    const editButton = result.getAllByRole('button', { name: 'button.modifica' });
```
Comment (AndreaCimini90) [inline packages/pn-personafisica-webapp/src/components/Contacts/__test__/CourtesyContacts.test.tsx:224]: don't use getAllByRole. check how it has been done in "override an existing email with a new one"

File: packages/pn-personafisica-webapp/src/components/Contacts/__test__/CourtesyContacts.test.tsx [hunk 4]
```diff
@@ -176,16 +213,16 @@ describe('CourtesyContacts Component', async () => {
       .reply(204);
     const result = render(
       <DigitalContactsCodeVerificationProvider>
-        <CourtesyContactItem type={CourtesyFieldType.PHONE} value={defaultPhoneAddress!.value} />
+        <CourtesyContacts contacts={digitalCourtesyAddresses} />
```
Comment (AndreaCimini90) [inline packages/pn-personafisica-webapp/src/components/Contacts/__test__/CourtesyContacts.test.tsx:216]: replace digitalCourtesyAddresses with [defaultPhoneAddress]

File: packages/pn-personafisica-webapp/src/components/Contacts/__test__/CourtesyContacts.test.tsx [hunk 5]
```diff
@@ -225,9 +262,23 @@ describe('CourtesyContacts Component', async () => {
       },
     ]);
     // simulate rerendering due to redux changes
+    const updatedDigitalCourtesyAddresses = [
```
Comment (AndreaCimini90) [inline packages/pn-personafisica-webapp/src/components/Contacts/__test__/CourtesyContacts.test.tsx:265]: this is

```
const updatedDigitalCourtesyAddresses = [
 {
   ...defaultPhoneAddress,
   value: +39 + phoneValue
 }
];
```

File: packages/pn-personafisica-webapp/src/components/Contacts/__test__/CourtesyContacts.test.tsx [hunk 6]
```diff
@@ -241,7 +292,11 @@ describe('CourtesyContacts Component', async () => {
     const phoneValue = defaultPhoneAddress!.value;
     const result = render(
       <DigitalContactsCodeVerificationProvider>
-        <CourtesyContactItem type={CourtesyFieldType.PHONE} value={phoneValue} />
+        <CourtesyContacts
+          contacts={digitalCourtesyAddresses.filter(
```
Comment (AndreaCimini90) [inline packages/pn-personafisica-webapp/src/components/Contacts/__test__/CourtesyContacts.test.tsx:296]: use [defaultPhoneAddress]

File: packages/pn-personafisica-webapp/src/components/Contacts/__test__/CourtesyContacts.test.tsx [hunk 7]
```diff
@@ -251,8 +306,8 @@ describe('CourtesyContacts Component', async () => {
     );
     const phoneText = result.getByText(phoneValue);
     expect(phoneText).toBeInTheDocument();
-    const deleteButton = result.getByRole('button', { name: 'button.elimina' });
-    fireEvent.click(deleteButton);
+    const deleteButton = result.getAllByRole('button', { name: 'button.elimina' });
```
Comment (AndreaCimini90) [inline packages/pn-personafisica-webapp/src/components/Contacts/__test__/CourtesyContacts.test.tsx:309]: Same as before

File: packages/pn-personafisica-webapp/src/components/Contacts/__test__/CourtesyContacts.test.tsx [hunk 8]
```diff
@@ -371,15 +426,16 @@ describe('CourtesyContacts Component', async () => {
       .reply(204);
     const result = render(
       <DigitalContactsCodeVerificationProvider>
-        <CourtesyContactItem type={CourtesyFieldType.EMAIL} value={emailValue} />
+        <CourtesyContacts contacts={digitalCourtesyAddresses} />
```
Comment (AndreaCimini90) [inline packages/pn-personafisica-webapp/src/components/Contacts/__test__/CourtesyContacts.test.tsx:429]: use [defaultEmailAddress]

File: packages/pn-personafisica-webapp/src/components/Contacts/__test__/CourtesyContacts.test.tsx [hunk 9]
```diff
@@ -434,17 +490,22 @@ describe('CourtesyContacts Component', async () => {
     const emailValue = defaultEmailAddress!.value;
     const result = render(
       <DigitalContactsCodeVerificationProvider>
-        <CourtesyContactItem type={CourtesyFieldType.EMAIL} value={emailValue} />
+        <CourtesyContacts
+          contacts={digitalCourtesyAddresses.filter(
```
Comment (AndreaCimini90) [inline packages/pn-personafisica-webapp/src/components/Contacts/__test__/CourtesyContacts.test.tsx:494]: use [defaultEmailAddress]

File: packages/pn-personagiuridica-webapp/src/components/Contacts/__test__/CourtesyContacts.test.tsx [hunk 10]
```diff
@@ -88,6 +88,41 @@ describe('CourtesyContacts Component', async () => {
     expect(emailInput).toBeInTheDocument();
   });
 
+  it('renders components - contacts', async () => {
```
Comment (AndreaCimini90) [inline packages/pn-personagiuridica-webapp/src/components/Contacts/__test__/CourtesyContacts.test.tsx:91]: Do the same changes request for pf

**Review by AndreaCimini90 — CHANGES_REQUESTED** [review]

File: packages/pn-personafisica-webapp/src/components/Contacts/__test__/CourtesyContacts.test.tsx (hunk 4 above)
Comment (SarahDonvito) [reply packages/pn-personafisica-webapp/src/components/Contacts/__test__/CourtesyContacts.test.tsx:216]: defaultPhoneAddress could be undefined, so I used [defaultPhoneAddress!]. Is that okay?

File: packages/pn-personafisica-webapp/src/components/Contacts/__test__/CourtesyContacts.test.tsx (hunk 5 above)
Comment (SarahDonvito) [reply packages/pn-personafisica-webapp/src/components/Contacts/__test__/CourtesyContacts.test.tsx:265]: phoneValue already contains +39 so I removed it in this object

File: packages/pn-personafisica-webapp/src/components/Contacts/__test__/CourtesyContacts.test.tsx (hunk 4 above)
Comment (AndreaCimini90) [reply packages/pn-personafisica-webapp/src/components/Contacts/__test__/CourtesyContacts.test.tsx:216]: yes

File: packages/pn-personagiuridica-webapp/src/components/Contacts/__test__/CourtesyContacts.test.tsx [hunk 11]
```diff
@@ -105,17 +140,18 @@ describe('CourtesyContacts Component', async () => {
...
-        <CourtesyContactItem type={CourtesyFieldType.PHONE} value="" />
+        <CourtesyContacts contacts={[]} />
       </DigitalContactsCodeVerificationProvider>
     );
     const input = result.container.querySelector(`[name="${CourtesyFieldType.PHONE}"]`);
     expect(input).toHaveValue('');
     fireEvent.change(input!, { target: { value: phoneValue } });
     await waitFor(() => expect(input!).toHaveValue(phoneValue));
-    const button = result.getByRole('button');
-    expect(button).toBeEnabled();
```
Comment (AndreaCimini90) [inline packages/pn-personagiuridica-webapp/src/components/Contacts/__test__/CourtesyContacts.test.tsx:116]: Why did you remove this check?

File: packages/pn-personagiuridica-webapp/src/components/Contacts/__test__/CourtesyContacts.test.tsx [hunk 12]
```diff
@@ -105,17 +140,19 @@ describe('CourtesyContacts Component', async () => {
...
     const input = result.container.querySelector(`[name="${CourtesyFieldType.PHONE}"]`);
     expect(input).toHaveValue('');
     fireEvent.change(input!, { target: { value: phoneValue } });
     await waitFor(() => expect(input!).toHaveValue(phoneValue));
-    const button = result.getByRole('button');
+    const button = result.getByTestId('courtesy-phone-button');
     expect(button).toBeEnabled();
     // save the phone
     fireEvent.click(button!);
+    await waitFor(() => expect(input!).toHaveValue(phoneValue));
```
Comment (AndreaCimini90) [inline packages/pn-personagiuridica-webapp/src/components/Contacts/__test__/CourtesyContacts.test.tsx:154]: this can be removed

**Review by AndreaCimini90 — CHANGES_REQUESTED** [review]

File: packages/pn-personafisica-webapp/src/components/Contacts/__test__/CourtesyContacts.test.tsx (hunk 6 above)
Comment (SarahDonvito) [reply packages/pn-personafisica-webapp/src/components/Contacts/__test__/CourtesyContacts.test.tsx]: I see it done, don't you?

File: packages/pn-personafisica-webapp/src/components/Contacts/__test__/CourtesyContacts.test.tsx (hunk 9 above)
Comment (AndreaCimini90) [reply packages/pn-personafisica-webapp/src/components/Contacts/__test__/CourtesyContacts.test.tsx:490]: @SarahDonvito this cr isn't done

File: packages/pn-personagiuridica-webapp/src/components/Contacts/__test__/CourtesyContacts.test.tsx (hunk 10 above)
Comment (AndreaCimini90) [reply packages/pn-personagiuridica-webapp/src/components/Contacts/__test__/CourtesyContacts.test.tsx:91]: Recheck changes done on pf

**Review by AndreaCimini90 — CHANGES_REQUESTED** [review]

**Review by AndreaCimini90 — APPROVED** [review]
